src/Layout/Presentation/presentation-utilities.ts

```diff
@@ -1,15 +1,35 @@
 import { PartyResult, DistrictResult, LevelingSeat, PartyRestQuotients } from "../../computation";
 import { roundNumber } from "../../utilities/number";
 
+function filterPartiesWithoutSeats(partyResults: PartyResult[], nationalResults?: PartyResult[]) {
+    let filteredResults = [...partyResults];
+
+    if (nationalResults) {
+        filteredResults = filteredResults.filter((party) => {
+            const nationalParty = nationalResults.find((nationalParty) => nationalParty.partyCode === party.partyCode);
+            if (!nationalParty) {
+                return false;
+            } else {
+                return nationalParty.totalSeats > 0;
+            }
+        });
+    } else {
+        filteredResults = filteredResults.filter((party) => party.totalSeats > 0);
+    }
+
+    return filteredResults;
+}
+
 export function getPartyTableData(
     partyResults: PartyResult[],
     showPartiesWithoutSeats: boolean,
-    numberOfDecimals: number
+    numberOfDecimals: number,
+    nationalResults?: PartyResult[],
 ): PartyResult[] {
     let filteredResults = [...partyResults];
 
     if (!showPartiesWithoutSeats) {
-        filteredResults = filteredResults.filter((party) => party.totalSeats > 0);
+        filteredResults = filterPartiesWithoutSeats(filteredResults);
```

Comment:
Why are we modifying this variable rather than just making a new one so we know exactly what it is in each state?

Reply:
Because if we stored the result into a second variable, we would essentially have to duplicate the majority of the rest of the function body in a second if statement, which uses one variable or the other depending on whether !showPartiesWithoutSeats is true.